query/physicalplan/filter.go

```diff
@@ -142,6 +142,52 @@ func binaryBooleanExpr(expr *logicalplan.BinaryExpr) (BooleanExpression, error)
 			Left:  left,
 			Right: right,
 		}, nil
+	case logicalplan.OpAdd, logicalplan.OpSub, logicalplan.OpMul, logicalplan.OpDiv:
```

Comment:
It still feels like we're creating this `ArithmeticExpr` at the wrong level of abstraction. The fact that the boolean expressions return a bitmap is an implementation detail abstracted out by the `columnProjection` interface. The boolean expressions are abstracted out by the `binaryExprProjection` implementation. I think what we should do is rename `binaryExprProjection` to `boolExprProjection` and add a new implementation of `columnProjection` that implements arithmetic expression evaluation (and is constructed with this arithmetic expr). Or something along those lines.

Reply:
Yes, that was my thought too. Ok, good to know this is the preferred way going forward. :relieved: